src/dropdown.ts

```diff
@@ -299,55 +314,54 @@ export default class Dropdown extends LitElement implements FormControl {
-    //
-    // If `#onOptionsSelectedChange()`, for example, weren't to bypass the setter, then
-    // our setting value below would cause `#onOptionsSelectedChange()` to be called
-    // again, which in turn would call this setter.
-    this.#value = value;
-
     if (!this.multiple && value.length > 1) {
       throw new Error('Only one value is allowed when not `multiple`.');
     }
 
+    this.#value = value;
+
     // When an option is selected, `#inputElementRef.value.value` is set the `label`
     // of the selected option. If Dropdown's `value` has been emptied, it means an
     // option is no longer selected. So the `value` of `#inputElementRef.value` should
     // be emptied too.
     if (
       !this.multiple &&
       this.value.length === 0 &&
-      this.#inputElementRef.value &&
-      this.selectedOptions.length === 1
+      this.#inputElementRef.value
     ) {
       this.#inputElementRef.value.value = '';
+      this.inputValue = '';
     }
 
-    for (const option of this.#optionElements) {
+    // `#onOptionsSelectedChange()` is called when an option is selected. It updates
+    // `this.selectedAndEnabledOptions`. Deselecting every option before reselecting
+    // those in `value` ensures tags appear in the same order as in `value`.
+    for (const option of this.selectedAndEnabledOptions) {
       this.#isSelectionFromValueSetter = true;
-
-      // If `value` is falsy, every option is left unselected. Otherwise, every
-      // `option.value` that's an empty string would be selected. If multiple
-      // options have the same `value`, they'll all be selected. No way to
-      // avoid that unfortunately.
-      option.selected = value.some((value) => value && value === option.value);
-
+      option.selected = false;
       this.#isSelectionFromValueSetter = false;
+    }
 
-      // We have a few options if `value` is set programmatically to include the
-      // value of a disabled option. We can throw, remove the value from `value`,
-      // or enable the option.
+    for (const optionValue of value) {
+      // We select only the first option with a given value so what the user sees as selected
+      // matches what is submitted with the form. If we were to select every matching option
+      // here, then the user would be under the impression that multiple instances of that
+      // value will be submitted with the form.
       //
-      // Throwing is attractive because the inclusion of a disabled option in `value`
-      // is likely a mistake, either due to bad data or developer error.
-      //
-      // But we only throw in development. So the form will be submitted with
-      // the new `value` in production regardless if it was by mistake. By enabling
-      // the option, we at least ensure the user is aware of the fact that it'll
-      // be included in the submission.
-      if (option.selected && option.disabled) {
-        option.disabled = false;
```

Comment:
Now handled centrally by `#onOptionsSelectedChange()`.  Moved the comment above there too.